wasm/src/api.rs

```diff
@@ -40,3 +53,82 @@ pub fn run_clvm(program: &[u8], args: &[u8]) -> Vec<u8> {
         Err(_eval_err) => format!("{:?}", _eval_err).into(),
     }
 }
+
+#[wasm_bindgen]
+/**
+ * Return serialized result of clvm program with cost
+ *
+ * cost will be available at the first 8 bytes of returned Vec<u8>(=Uint8Array).
+ * bytes of node will be available from 8th byte offset of returned Vec<u8>.
+ *
+ * @example
+ * const result = run_chia_program(...); // Uint8Array
+ * const cost = vec_u8_to_u64(result.subarray(0, 8)) // BigInt;
+ * const serialized_node = result.subarray(8); // Uint8Array
+ */
+pub fn run_chia_program(
+    program: &[u8],
+    args: &[u8],
+    max_cost: Cost, // Expecting `BigInt` to be passed from JavaScript world
+    flag: u32,
+) -> Vec<u8> {
+    let mut allocator = Allocator::new();
+    let program = node_from_bytes(&mut allocator, program).unwrap();
+    let args = node_from_bytes(&mut allocator, args).unwrap();
+    let dialect = ChiaDialect::new(flag);
+
+    let r = run_program(
+        &mut allocator,
+        &dialect,
+        program,
+        args,
+        max_cost,
+        None,
+    );
+    match r {
+        Ok(reduction) => {
+            let cost = reduction.0;
+            let node_bytes = node_to_bytes(
+                &Node::new(&allocator, reduction.1)
+            ).unwrap();
+
+            [
+                u64_to_vec_u8(cost),
+                node_bytes,
+            ].concat()
+        },
+        Err(_eval_err) => format!("{:?}", _eval_err).into(),
+    }
+}
+
+#[wasm_bindgen]
+pub fn u64_to_vec_u8(num: u64) -> Vec<u8> {
+    let u8_vec = vec![
+        ((num & 0xff000000_00000000) >> 56) as u8,
+        ((num & 0x00ff0000_00000000) >> 48) as u8,
+        ((num & 0x0000ff00_00000000) >> 40) as u8,
+        ((num & 0x000000ff_00000000) >> 32) as u8,
+        ((num & 0xff000000) >> 24) as u8,
+        ((num & 0x00ff0000) >> 16) as u8,
+        ((num & 0x0000ff00) >> 8) as u8,
+        ((num & 0x000000ff) >> 0) as u8,
+    ];
+
+    u8_vec
+}
+
+#[wasm_bindgen]
+pub fn vec_u8_to_u64(nums: &[u8]) -> u64 {
+    let u64_value =
```

Comment:
this function could be:
```
u64::from_be_bytes(nums).expect("not enough bytes")
```

Also, there's no need to create a variable to return, just omit the `;` at the end.

Reply:
In my environment, this didn't work.
```rust
#[wasm_bindgen]
pub fn vec_u8_to_u64(nums: &[u8]) -> u64 {
    u64::from_be_bytes(nums).expect("not enough bytes")
}
```

So I tried this one:
```rust
#[wasm_bindgen]
pub fn vec_u8_to_u64(nums: &[u8]) -> u64 {
    u64::from_be_bytes(nums.try_into().expect("not enough bytes"))
}
```